### PR TITLE
Trajectory manipulation improvements

### PR DESCRIPTION
**Pull request by eric-wieser**
This makes a couple of fixes, and adds `trajcat` to overlay trajectories.

This is defined such that the second trajectory always overwrites the first, unless the second trajectory extends to infinity over the first, in which case the first overwrites the second.

I needed this in order to `vertcat` two trajectories with zero-padding:

	tspan = [0 max(utrajs{1}.tspan(2), utrajs{2}.tspan(2))];
	
	for i=1:length(utrajs)
		u0 = zeros(plant.getNumInputs, 1);
		utrajs{i} = trim(trajcat(u0, utrajs{i}, u0), tspan);
		xtrajs{i} = trim(trajcat(x0, xtrajs{i}, xf), tspan);
	end
	utraj = vertcat(utrajs{:});
	xtraj = vertcat(xtrajs{:});

This brought up some frame-dropping within the api.

### CODE REVIEW

**Comment by eric-wieser** [comment]
I've now realized that `trajcat` is a little similar to `HybridTrajectory`...

**Comment by RussTedrake** [comment]
actually it's closer to (identical to?) the append method of the PPTrajectory class.  If trajcat is a generalization, then it should match the name.

File: drake/systems/trajectories/Trajectory.m [hunk 1]
```diff
@@ -115,6 +115,30 @@
         breaks = breaks(breaks>=tspan(1) & breaks<=tspan(2));
         breaks = breaks';
     end
+    function [a,b,breaks,thandover] = setupTrajectoryChain(a,b)
```
Comment (RussTedrake) [inline drake/systems/trajectories/Trajectory.m]: document this method

File: drake/systems/trajectories/Trajectory.m [hunk 2]
```diff
@@ -148,8 +172,31 @@
...
           error('dimensions 1 and 3:end must match');
         end
+        fr = getOutputFrame(c);
         c = FunctionHandleTrajectory(@(t) horzcat(c.eval(t),b.eval(t)),[cdim(1),cdim(2)+bdim(2),cdim(3:end)],breaks);
-        c = setOutputFrame(c,MultiCoordinateFrame({getOutputFrame(c),getOutputFrame(b)}));
+        c = setOutputFrame(c,MultiCoordinateFrame({fr,getOutputFrame(b)}));
+      end
+    end
+
+    function c = trajcat(a, varargin)
```
Comment (RussTedrake) [inline drake/systems/trajectories/Trajectory.m]: rename to append (if it does generalize PPTrajectory).
and add a simple unit test for coverage

**Comment by eric-wieser** [comment]
For now, I've split out the non-concatenation parts of this PR into #1500

**Comment by RussTedrake** [comment]
sounds good.  will close this one.

**Pull request closed without merging**